README.md

```diff
@@ -1,234 +1,679 @@
+    - [The `Label` schema](#the-label-schema)
+- [Organization on Disk](#organization-on-disk)
+    - [Organization of task labels](#organization-of-task-labels)
+- [Validation](#validation)
+- [Example: MIMIC-IV demo dataset](#example-mimic-iv-demo-dataset)
+
+## Philosophy
+
+At the heart of MEDS is a simple yet powerful idea: nearly all EHR data can be modeled as a minimal tuple. We
+believe that the essence of a clinical event or measurement can be effectively described using three core
+components:
+
+1. _subject_: The primary entity for which care observations are recorded. Typically, this is an individual
+    with a complete sequence of observations. In some datasets (e.g., eICU), a subject may refer to a single
+    hospital admission rather than the entire individual record.
+
+2. _time_: The time that a measurement was observed.
+
+3. _code_: The descriptor of what measurement is being observed.
+
+This minimalist representation captures the essence of EHR data while providing a consistent foundation for
+further analysis and enrichment.
+
+> [!NOTE]
+> In this documentation, we will primarily use the term "measurement" to refer to a single observation about a
+> subject at a given time (i.e., a row of MEDS data). We may use the term "event" to refer to this as well, or
+> to refer to all measurements that occur at a unique point in time, depending on context.
+
+## The Schemas
+
+Building on this philosophy, MEDS defines five primary schema components:
+
+| **Component**     | **Description**                                                                                                                       | **Implementation** |
+| ----------------- | ------------------------------------------------------------------------------------------------------------------------------------- | ------------------ |
+| `Data`            | Describes the core medical data, organized as sequences of subject observations.                                                      | PyArrow            |
+| `DatasetMetadata` | Captures metadata about the source dataset, including its name, version, and details of its conversion to MEDS (e.g., ETL details).   | JSON               |
+| `CodeMetadata`    | Provides metadata for the codes used to describe the types of measurements observed in the dataset.                                   | PyArrow            |
+| `SubjectSplit`    | Stores information on how subjects are partitioned into subpopulations (e.g., training, tuning, held-out) for machine learning tasks. | PyArrow            |
+| `Label`           | Defines the structure for labels that may be predicted about a subject at specific times in the subject record.                       | PyArrow            |
+
+Below, each schema is introduced in detail. Usage examples and a practical demonstration with the [MIMIC-IV demo](https://physionet.org/content/mimic-iv-demo/2.2/) dataset are provided in a later section.
```

Comment:
I think it would be helpful to clarify that a single MEDS dataset comprises:

- Exactly one DatasetMetadata file 
- Exactly one CodeMetadata file
- One or many Data files
- ...

Perhaps worth its own section?